great_expectations/core/id_dict.py

```diff
@@ -18,39 +22,50 @@ def to_id(self, id_keys=None, id_ignore_keys=None):
             key = list(id_keys)[0]
             return f"{key}={str(self[key])}"
 
-        _id_dict = {k: self[k] for k in id_keys}
+        _id_dict = convert_to_json_serializable(data={k: self[k] for k in id_keys})
         return hashlib.md5(
             json.dumps(_id_dict, sort_keys=True).encode("utf-8")
         ).hexdigest()
 
-    @staticmethod
-    def convert_dictionary_to_id_dict(data: Optional[Any]):
-        """
-        This method converts any nested "data" argument of dictionary or iterable type to "IDDict" object (recursively).
-        """
-        if isinstance(data, dict):
-            data = IDDict(data)
-
-            key: str
-            value: Any
-            for key, value in data.items():
-                data[key] = IDDict.convert_dictionary_to_id_dict(data=value)
-        elif isinstance(data, (list, set, tuple)):
-            data_type: type = type(data)
-
-            value: Any  # type: ignore[no-redef]
-            data = data_type(
-                [IDDict.convert_dictionary_to_id_dict(data=value) for value in data]
-            )
-
-        return data
-
     def __hash__(self) -> int:  # type: ignore[override]
         """Overrides the default implementation"""
         _result_hash: int = hash(self.to_id())
         return _result_hash
 
 
+def convert_dictionary_to_id_dict(source: Union[T, dict]) -> Union[T, IDDict]:
+    if isinstance(source, dict):
+        return _convert_dictionary_to_id_dict(source=IDDict(source))
+
+    # Must allow for non-dictionary source types, since their internal nested structures may contain dictionaries.
```

Comment:
Does that mean this method is inappropriately named?

Reply:
@NathanFarmer Potentially; I just could not come up with a better name; at least this name indicates the starting (top-level) name of a potentially nested structure -- the first call typically comes with correctly-named structure; the recursion can encounter ones that do not match this name.  Any ideas are welcome (or my comment can suffice).  Thanks!

Reply:
`convert_iterable_dictionaires_to_id_dict`? Yes, a `str` is also an iterable, but a `str` cannot contain a `dict`.